File: app/src/main/java/com/jameskbride/eventbusmvpdemo/main/OrdersAdapterFactory.java
package com.jameskbride.eventbusmvpdemo.main;

import android.content.Context;
import android.support.annotation.LayoutRes;
import android.widget.ArrayAdapter;

import java.util.List;

class OrdersAdapterFactory {

    public ArrayAdapter<String> make(Context context, @LayoutRes int layoutId, List<String> descriptions) {
        return new ArrayAdapter(context, layoutId, descriptions);
    }
}

File: app/src/main/java/com/jameskbride/eventbusmvpdemo/main/OrdersAdapterFactory.kt
package com.jameskbride.eventbusmvpdemo.main

import android.content.Context
import android.widget.ArrayAdapter
import androidx.annotation.LayoutRes

class OrdersAdapterFactory {

    fun make(context: Context, @LayoutRes layoutId:Int, descriptions:List<String>):ArrayAdapter<String> {
        return ArrayAdapter(context, layoutId, descriptions)
    }
}